Can't package with `mvn package`

Seems like `${session.executionRootDirectory}` isn't substituted with whatever it's meant to be.
`mvn package -Dmaven.test.skip=true`
...
```console
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:2.3:single (make-assembly) on project jzmq-jni: Failed to create assembly: Error adding file to archive: /home/build/jzmq/jzmq-jni/${session.executionRootDirectory}/jzmq-jni/src/main/c++/.libs/libjzmq.so isn't a file. -> [Help 1]
```

Same issue I am facing too 
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:2.3:single (make-assembly) on project jzmq-jni: Failed to create assembly: Error adding file to archive: /usr/lib/jzmq/jzmq-jni/${session.executionRootDirectory}/jzmq-jni/src/main/c++/.libs/libjzmq.so isn't a file.

--- a/jzmq-core/pom.xml
+++ b/jzmq-core/pom.xml
@@ -19,4 +19,13 @@
       <version>3.1.1-SNAPSHOT</version>
     </dependency>
   </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.commonjava.maven.plugins</groupId>
+        <artifactId>directory-maven-plugin</artifactId>
+      </plugin>
+    </plugins>
+  </build>
 </project>

--- a/jzmq-devices/pom.xml
+++ b/jzmq-devices/pom.xml
@@ -19,4 +19,13 @@
       <version>3.1.1-SNAPSHOT</version>
     </dependency>
   </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.commonjava.maven.plugins</groupId>
+        <artifactId>directory-maven-plugin</artifactId>
+      </plugin>
+    </plugins>
+  </build>
 </project>

--- a/jzmq-jni/pom.xml
+++ b/jzmq-jni/pom.xml
@@ -52,6 +52,10 @@
   <build>
     <plugins>
       <plugin>
+        <groupId>org.commonjava.maven.plugins</groupId>
+        <artifactId>directory-maven-plugin</artifactId>
+      </plugin>
+      <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-gpg-plugin</artifactId>
         <executions>

--- a/pom.xml
+++ b/pom.xml
@@ -64,8 +64,8 @@
       </activation>
       <properties>
         <native.os>${os.name}</native.os>
-        <native.path>${session.executionRootDirectory}/jzmq-jni/src/main/c++/.libs/libjzmq.so</native.path>
-        <native.library-path>${session.executionRootDirectory}/jzmq-jni/src/main/c++/.libs/</native.library-path>
+        <native.path>${root.directory}/jzmq-jni/src/main/c++/.libs/libjzmq.so</native.path>
+        <native.library-path>${root.directory}/jzmq-jni/src/main/c++/.libs/</native.library-path>
       </properties>
     </profile>
     <profile>
@@ -77,8 +77,8 @@
       </activation>
       <properties>
         <native.os>Windows</native.os>
-        <native.path>${session.executionRootDirectory}/jzmq-jni/lib/jzmq.dll</native.path>
-        <native.library-path>${session.executionRootDirectory}/jzmq-jni/lib</native.library-path>
+        <native.path>${root.directory}/jzmq-jni/lib/jzmq.dll</native.path>
+        <native.library-path>${root.directory}/jzmq-jni/lib</native.library-path>
       </properties>
     </profile>
     <profile>
@@ -90,8 +90,8 @@
       </activation>
       <properties>
         <native.os>${os.name}</native.os>
-        <native.path>${session.executionRootDirectory}/jzmq-jni/src/main/c++/.libs/libjzmq.dylib</native.path>
-        <native.library-path>${session.executionRootDirectory}/jzmq-jni/src/main/c++/.libs/</native.library-path>
+        <native.path>${root.directory}/jzmq-jni/src/main/c++/.libs/libjzmq.dylib</native.path>
+        <native.library-path>${root.directory}/jzmq-jni/src/main/c++/.libs/</native.library-path>
       </properties>
     </profile>
     <profile>
@@ -119,6 +119,23 @@
   <build>
     <pluginManagement>
       <plugins>
+        <plugin>
+          <groupId>org.commonjava.maven.plugins</groupId>
+          <artifactId>directory-maven-plugin</artifactId>
+          <version>0.1</version>
+          <executions>
+            <execution>
+              <id>directories</id>
+              <goals>
+                <goal>highest-basedir</goal>
+              </goals>
+              <phase>initialize</phase>
+              <configuration>
+                <property>root.directory</property>
+              </configuration>
+            </execution>
+          </executions>
+        </plugin>
         <plugin>
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-gpg-plugin</artifactId>

